openedx/features/idea/models.py

```diff
@@ -2,5 +2,69 @@
 from __future__ import unicode_literals
 
 from django.db import models
+from django.contrib.auth.models import User
+from django.core.validators import FileExtensionValidator
+from django_countries.fields import CountryField
 
-# Create your models here.
+from functools import partial
+
+from lms.djangoapps.onboarding.models import Organization
+from .helpers import upload_to_path
+from .constants import CITY_MAX_LENGTH, TITLE_MAX_LENGTH, OVERVIEW_MAX_LENGTH
+
+
+class Location(models.Model):
+    country = CountryField()
+    city = models.CharField(max_length=CITY_MAX_LENGTH)
+
+    @property
+    def location(self):
+        return '{city}, {country}'.format(city=self.city, country=self.country.name)
+
+    class Meta:
+        abstract = True
+
+
+class VisualAttachment(models.Model):
+
+    video_link = models.URLField(blank=True, null=True)
+    image = models.ImageField(
+        upload_to=partial(upload_to_path, folder='images'), blank=True, null=True,
+        validators=[FileExtensionValidator(['jpg', 'png'])],
+        help_text='Accepted extensions: .jpg, .png'
+    )
+    file = models.FileField(
+        upload_to=partial(upload_to_path, folder='files'), blank=True, null=True,
+        validators=[FileExtensionValidator(['docx', 'pdf', 'txt'])],
+        help_text='Accepted extensions: .docx, .pdf, .txt'
+    )
+
+    class Meta:
+        abstract = True
+
+
+class OrganizationBase(models.Model):
+    organization = models.ForeignKey(
+        Organization,
+        related_name='%(app_label)s_%(class)ss',
+        related_query_name='%(app_label)s_%(class)s',
+        on_delete=models.CASCADE
+    )
+    organization_mission = models.TextField()
+
+    class Meta:
+        abstract = True
```

Comment:
these classes should be moved to some common place. 
Let's please discuss with @ahmed-faraz46 @saad-waqar for suitable place.

Reply:
Lets discuss this and defer it until marketplace